test/cmd/projects.sh

```diff
@@ -35,7 +35,39 @@ os::cmd::expect_success_and_not_text 'oc config view -o jsonpath="{.contexts[*].
+os::cmd::expect_success_and_text 'oc login -u system:admin -n default' 'Login successful.|system:admin' # login as admin
+# test that admin user can edit all fields regardless of which endpoint they go through
+os::cmd::expect_success_and_text 'oc annotate project nstestproj key3=val3 --overwrite' 'project "nstestproj" annotated'
+os::cmd::expect_success_and_text 'oc annotate namespace nstestproj key4=val4 --overwrite' 'namespace "nstestproj" annotated'
+os::cmd::expect_success_and_text 'oc login -u nstest -p nstest' 'Login successful.' # login standard user again
+# test that standard user still cannot edit immutable fields
+os::cmd::expect_failure_and_text 'oc annotate project nstestproj key1=val1 --overwrite' 'The Project "nstestproj" is invalid: metadata.annotations\[key1\]: Invalid value: "val1": field is immutable, try updating the namespace'
+os::cmd::expect_failure_and_text 'oc annotate namespace nstestproj key2=val2 --overwrite' 'Error from server: User "nstest" cannot "patch" "namespaces" with name "nstestproj" in project "nstestproj"'
+# this will now be successful because there is no diff between the old and new project
+os::cmd::expect_success_and_text 'oc annotate project nstestproj key3=val3 --overwrite' 'project "nstestproj" annotated'
+# this fails early at the authorization stage even though there is no diff
+os::cmd::expect_failure_and_text 'oc annotate namespace nstestproj key4=val4 --overwrite' 'Error from server: User "nstest" cannot "patch" "namespaces" with name "nstestproj" in project "nstestproj"'
+# same tests as above but with labels instead of annotations
+os::cmd::expect_failure_and_text 'oc label project nstestproj keyA=valA --overwrite' 'The Project "nstestproj" is invalid: metadata.labels\[keyA\]: Invalid value: "valA": field is immutable, try updating the namespace'
+os::cmd::expect_failure_and_text 'oc label namespace nstestproj keyB=valB --overwrite' 'Error from server: User "nstest" cannot "patch" "namespaces" with name "nstestproj" in project "nstestproj"'
+os::cmd::expect_success_and_text 'oc login -u system:admin -n default' 'Login successful.|system:admin'
+os::cmd::expect_success_and_text 'oc label project nstestproj keyC=valC --overwrite' 'project "nstestproj" labeled'
+os::cmd::expect_success_and_text 'oc label namespace nstestproj keyD=valD --overwrite' 'namespace "nstestproj" labeled'
+os::cmd::expect_success_and_text 'oc login -u nstest -p nstest' 'Login successful.'
+os::cmd::expect_failure_and_text 'oc label project nstestproj keyA=valA --overwrite' 'The Project "nstestproj" is invalid: metadata.labels\[keyA\]: Invalid value: "valA": field is immutable, try updating the namespace'
+os::cmd::expect_failure_and_text 'oc label namespace nstestproj keyB=valB --overwrite' 'Error from server: User "nstest" cannot "patch" "namespaces" with name "nstestproj" in project "nstestproj"'
+os::cmd::expect_success_and_text 'oc label project nstestproj keyC=valC --overwrite' 'project "nstestproj" not labeled'
```

Comment:
`oc label` (unlike `oc annotate`) checks to see if there is a diff between the two objects when printing the success message.